Dockerfile

```diff
@@ -1,7 +1,20 @@
-FROM gliderlabs/alpine
+FROM golang:1.8.3 as builder
+
+ARG VERSION=0.0.1
+WORKDIR /go/src/github.com/oliver006/redis_exporter
+
+COPY . .
+
+RUN go get -d -v .
+RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo  -ldflags "-X main.VERSION=${VERSION} -X main.BUILD_DATE=$(date +%F-%T) -X main.COMMIT_SHA1=`git rev-parse HEAD`" -o redis_exporter .
```

Comment:
Question: What's the benefit of using the builder container as opposed to building the binary separately in the CircleCI runtime and the `COPY` it into the container? 

We now have the build config (LD_FLAGS, Golang version) in two places which I'd prefer to avoid.

Reply:
Hi @oliver006,

Sorry for late response. 

- First of all, I had an issue with golang version on my local machine(golang 1.9.1). I had failed to compile redis_exporter on my laptop. But, multi-stage-build was a way to resolve that kinds of version or dependency issues on the machine.
- Anyone can build new update on local machine and test it easy.

Frankly, it doesn't matter whether or not using multi-stage-build. But, it is one of benefits of why we are using docker system.

FYI, you can find multi-stage-build at https://docs.docker.com/engine/userguide/eng-image/multistage-build/#use-multi-stage-builds.

Please let me know whether or not you want to keep the code.


Best regards,

Chang

Reply:
I see, thanks for explaining. Considering that the "real" builds are done in a fresh container in CircleCi I'd prefer to keep the Dockerfile the way it is right now and build on the CI host.